helpers/calendar/all_analytics_helper.py

```diff
@@ -67,43 +64,70 @@ def get_all_analytics(self, query, start_date, end_date, location_id, unconverte
         rooms = query.filter_by(state="active", location_id=location_id).all()
         room_analytics = []
         bookings = 0
+        total_checkins = 0
+        total_app_bookings = 0
+        total_auto_cancellations = 0
+        total_cancellations = 0
+        bookings_count = []
+        all_events = []
         for room in rooms:
+            events = []
+            duaration_in_minutes = 0
             cancellations = 0
             checkins = 0
-            events = []
+            auto_cancellations = 0
+            app_bookings = 0
             try:
                 events_result = CommonAnalytics.get_all_events_in_a_room(
                     self, room.id, start_date, end_date)
             except GraphQLError:
                 continue
-            bookings_count = AllAnalyticsHelper.bookings_count(self, unconverted_dates, events_result) # noqa
+            all_events += events_result
             bookings += len(events_result)
             for event in events_result:
                 if event.checked_in:
                     checkins += 1
+                if event.app_booking:
+                    app_bookings += 1
+                if event.auto_cancelled:
+                    auto_cancellations += 1
                 if event.cancelled:
                     cancellations += 1
-                duaration_in_minutes = CommonAnalytics.get_time_duration_for_event(self, event.start_time, event.end_time) # noqa
-                current_event = Event(
-                    duration_in_minutes=duaration_in_minutes)
-                events.append(current_event)
+                duaration_in_minutes += CommonAnalytics.get_time_duration_for_event( # noqa
+                    self, event.start_time, event.end_time
+                )
+            current_event = Event(
+                duration_in_minutes=duaration_in_minutes)
+            events.append(current_event)
+            total_checkins += checkins
+            total_app_bookings += app_bookings
+            total_auto_cancellations += auto_cancellations
+            total_cancellations += cancellations
+            num_of_events = len(events_result)
             room_analytic = {
-                'number_of_meetings': len(events_result),
+                'number_of_meetings': num_of_events,
                 'room_name': room.name,
+                'num_of_events': num_of_events,
```

Comment:
@kwanj-k Why are there `number of meetings` and `num_of_events` since they are both pointing to the same variable?

Reply:
We decided to declare the num_of_events variable to reuse it in that loop, num_of_events will be used to calculate the percentage share.